fix: updated regex on validateNetwork to support special characters.

alessandrocapanna opened the pull request:
Fix issue #17614 

Now the regex support special characters as mentioned on issue.

Test made with faked credential: "http://user:fake%21password@someproxy.com:80"

Before:
<img width="473" alt="Schermata 2024-02-13 alle 01 16 07" src="https://github.com/kubernetes/minikube/assets/63300338/d6fe05a6-1b6f-4962-8868-27ddd77f4cd2">

After:

<img width="462" alt="Schermata 2024-02-13 alle 00 58 13" src="https://github.com/kubernetes/minikube/assets/63300338/9ac42260-ed78-429b-a596-89de9ae86bd1">

alessandrocapanna commented:
fixed also if password have the '@'

spowelljr commented:
I agree with @Zhongyi-Lu, using his example string `http://jdoe:%n0tRe@alPassword!@minikube.sigs.k8s.io:8080`

Using `//(\w+):.*?@` matches `//jdoe:%n0tRe@`

While `//(\w+):.*@` matches `//jdoe:%n0tRe@alPassword!@`

alessandrocapanna commented:
Yeah... I also agree with @Zhongyi-Lu, I will modify the pull request, I will also insert strings like those suggested in the tests :D

spowelljr commented:
Hi @alessandrocapanna, I just realized we have a function somewhere else in our code that does this as well.

`maskProxyPassword` in `pkg/minikube/node/config.go`

Could you make the function pulic and move it to `pkg/util`, then update any refernces to, and then we can make the code you modified to use the function as well. There are also a test for the function in `pkg/minikube/node/config_test.go` you could move as well and add additional test cases if needed. Feel free to chance the function if it's not worked as expected.

alessandrocapanna commented:
Hi @spowelljr.
I've completed the refactor you suggested. I split the maskProxyPassword function into two separate functions (under `pkg/util`) and made the necessary corrections. Additionally, I updated the references accordingly and added tests for both functions.

spowelljr commented:
/ok-to-test

spowelljr reviewed (approved):
Thanks for the PR!
